src/node.rs

```diff
@@ -135,8 +135,12 @@ where
             }
             .boxed()
         })?
-        .get("sync_status", |_req, state| {
-            async move { Ok(state.sync_status().await?) }.boxed()
+        .at("sync_status", |_req, state| {
+            async move {
+                let fut = state.read(|state| state.sync_status()).await;
```

Comment:
ok okay. I think its to avoid read lock here

Reply:
Oh yeah, in terms of why are we returning a future at all as opposed to just returning a result, that is why.